### PR TITLE
Feature/F15 & F17get user roles and based on role add permissions

### PR DESCRIPTION
**Pull request by AntonioNik**
Screenshot 2023-07-05 at 13 12 11

### CODE REVIEW

File: Frontend/src/store/loggedUser/loggedUserInterface.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface LoggedUser {
+  currentUser: {
+    id: number;
```
Comment (betinabp) [inline Frontend/src/store/loggedUser/loggedUserInterface.ts:3]: `export interface CurrentUser {
id: number;
  firstName: string;
    lastName: string;
    email: string; }` and then in this interface `currentUser: CurrentUser`. Avoid adding nested object in the interface, instead create another interface for them

File: Frontend/src/services/fetchWrapper.ts [hunk 2]
```diff
@@ -31,15 +34,20 @@ const _delete = async (url: string) => {
...
-const handleResponse = async (response: AxiosResponse) => {
-  if (response.status !== 200) {
-    const error =
-      (response.data && response.data.message) || response.statusText;
-    throw error;
-  }
-
-  return response.data;
-};
+// const handleResponse = (response: AxiosResponse) => {
```
Comment (betinabp) [inline Frontend/src/services/fetchWrapper.ts:37]: Why this is commented out?

File: Frontend/src/services/authService.ts [hunk 3]
```diff
@@ -13,5 +13,12 @@ const login = async (body: User) => {
     throw error;
   }
 };
+const getUserRole = async (token: string) => {
+  const response = await fetchWrapper.get(
```
Comment (betinabp) [inline Frontend/src/services/authService.ts:17]: The token should not be passed as an argument. 
Here in the service you should do ` const token = localStorage.getItem("token");`, not in the component!

File: Frontend/src/pages/login/ErrorStyles.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.errorMessage {
+  color: rgb(182, 73, 73);
```
Comment (betinabp) [inline Frontend/src/pages/login/ErrorStyles.scss:2]: Move the color as a variable

File: Frontend/src/components/navbar/Navbar.tsx [hunk 5]
```diff
@@ -65,6 +69,13 @@ const Navbar = ({ currentUser, manager }: NavbarProps) => {
     setAnchorElUser(null);
   };
 
+  const handleLogout = (event: React.MouseEvent<HTMLLIElement>) => {
+    dispatch(logout());
+    handleCloseUserMenu();
+    navigate("/login");
+    window.location.reload();
```
Comment (betinabp) [inline Frontend/src/components/navbar/Navbar.tsx:76]: Why do you reload the screen?

**Review by betinabp — CHANGES_REQUESTED** [review]

File: Frontend/src/services/fetchWrapper.ts (hunk 2 above)
Comment (AntonioNik) [reply Frontend/src/services/fetchWrapper.ts:37]: i did not work out as i was expecting it to and commented it out to test the other simple approach..and it worked out

**Pull request merged**